### PR TITLE
Fix MQTTC to wait for connection (MQTT ACK)

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Fix MQTTC to wait for connection (MQTT ACK) and avoid passing '\0' in the payload
## Impact
Users will get MQTTC example working with other MQTT Brokers like TagoIO
## Testing
ESP32-Devkitc sending data to TagoIO

### CODE REVIEW

**Comment by acassis** [comment]
Please ignore the CI nxstyle check, the MQTT-C uses CamelCase like "MQTTErrors"

**Review by pkarashchenko — APPROVED** [review]

File: netutils/mqttc/0001_add_connection_status.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ /**
+diff -Naur MQTT-C_orig/src/mqtt.c MQTT-C/src/mqtt.c
+--- MQTT-C_orig/src/mqtt.c	2021-03-29 14:53:52.000000000 -0300
++++ MQTT-C/src/mqtt.c	2023-05-12 19:51:38.675627411 -0300
+@@ -732,6 +732,8 @@
+                         mqtt_recv_ret = MQTT_ERROR_CONNECTION_REFUSED;
+                     }
+                     break;
++                } else {
++                    client->event_connect = true;
```
Comment (xiaoxiang781216) [inline netutils/mqttc/0001_add_connection_status.patch:22]: but should we clear this flag in mqtt_init(), mqtt_init_reconnect() and mqtt_reinit()?

Comment (acassis) [reply netutils/mqttc/0001_add_connection_status.patch:22]: I did exactly how Pieter implemented here:
https://github.com/LiamBindle/MQTT-C/issues/163

But I agree, it is better to clear this flag in other places, I'll update the patch

File: examples/mqttc/mqttc_pub.c [hunk 2]
```diff
@@ -278,10 +279,22 @@ int main(int argc, FAR char *argv[])
       goto err_with_socket;
     }
 
+  /* Wait for MQTT ACK or time-out */
+
+  while (!mqtt_cfg.client.event_connect && --timeout > 0)
+   {
```
Comment (pkarashchenko) [inline examples/mqttc/mqttc_pub.c:285]:
```suggestion
    {
```

File: examples/mqttc/mqttc_pub.c [hunk 3]
```diff
@@ -278,10 +279,22 @@ int main(int argc, FAR char *argv[])
...
+     usleep(10000);
+   }
```
Comment (pkarashchenko) [inline examples/mqttc/mqttc_pub.c:287]:
```suggestion
    }
```

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**